### PR TITLE
Update release documentation

### PR DESCRIPTION
**Pull request by KrishnaIyer**
#### Summary

Just restructured the release section of `DEVELOPMENT.md`

#### Changes

- Restructure the section.

#### Notes for Reviewers

Since the diff can be a bit trick to check, please review the steps as though it's a new doc.

#### Checklist

- [ ] Scope: The referenced issue is addressed, there are no unrelated changes.
- [ ] Compatibility: The changes are backwards compatible with existing API, database and configuration, according to the stability commitments in `README.md`.
- [ ] Testing: The changes are covered with unit tests. The changes are tested manually as well.
- [x] Documentation: Relevant documentation is added or updated.
- [ ] Changelog: Significant features, behavior changes, deprecations and fixes are added to `CHANGELOG.md`.
- [x] Commits: Commit messages follow guidelines in `CONTRIBUTING.md`, there are no fixup commits left.

### CODE REVIEW

**Comment by KrishnaIyer** [comment]
Hmm ok. But if you look at the table, there are 11 steps and 8 of them are common "actions" with different base/target branches. We could of course just add 2 different sections but that would have some repetition.

**Comment by johanstokking** [comment]
In my personal opinion this is less readable than two verbose sections.

@htdvisser what do you think?

**Comment by htdvisser** [comment]
I agree with @johanstokking that this doesn't make it more readable and that it's better to use two sections. A little bit of copy-paste isn't a problem for me. When I start the release process, I just want to go to the appropriate section and follow the steps.

**Comment by KrishnaIyer** [comment]
Ok. I've updated it.

File: DEVELOPMENT.md [hunk 1]
```diff
@@ -462,10 +462,84 @@ This will compile binaries for all supported platforms, `deb`, `rpm` and Snapcra
 > Note: The operating system and architecture represent the name of the directory in `dist` in which the binaries are placed.
 > For example, the binaries for Darwin x64 (macOS) will be located at `dist/darwin_amd64`.
 
-Releasing a new version consists of the following steps:
+A new version can be released from the `master` branch or a `backport` branch. The necessary steps for each are detailed below.
 
-1. Creating a `release/<version>` branch off `master` or `backport/<minor>` (further, called "release branch") (e.g. `release/v3.2.1` or `backport/v3.2`).
-2. Updating the `CHANGELOG.md` file:
+> Note: To get the target version, you can run `version=$(./mage version:bumpXXX version:current)`, where xxx is the type of new release (minor/patch/RC).
```
Comment (johanstokking) [inline DEVELOPMENT.md]: This is explained below; probably move that section up

File: DEVELOPMENT.md [hunk 2]
```diff
@@ -486,71 +560,65 @@ Releasing a new version consists of the following steps:
...
-$ git checkout <target> # master or backport/<minor>
-$ git merge --ff-only "release/${version}"
-$ ./mage version:bumpPatch version:tag
-$ git push origin ${version}
-$ git push origin <target>
+$ ./mage version:bumpXXX version:tag
 ```
 
-After pushing the tag, our CI system will start building the release. When this is done, you'll find a new release on the [releases page](https://github.com/TheThingsNetwork/lorawan-stack/releases). After this is done, you'll need to edit the release notes. We typically copy-paste these from `CHANGELOG.md`.
+For RCs, Make sure to use the same bumping combination (ex: `verion:bumpXXX version:bumpYYY`) as used in the bump step above.
```
Comment (johanstokking) [inline DEVELOPMENT.md]: Typo

File: DEVELOPMENT.md [hunk 3]
```diff
@@ -486,71 +560,65 @@ Releasing a new version consists of the following steps:
...
+
+
```
Comment (johanstokking) [inline DEVELOPMENT.md]: Double empty line

File: DEVELOPMENT.md [hunk 4]
```diff
@@ -486,71 +560,65 @@ Releasing a new version consists of the following steps:
...
+### Docker Latest Tag
 
 When the CI system pushed the Docker image, it gets tagged as the current minor and patch version. If this release is not a backport but a latest stable release, you should manually tag and push `latest`:
 
-```bash
+```
```
Comment (johanstokking) [inline DEVELOPMENT.md]: Why not bash

File: DEVELOPMENT.md (hunk 1 above)
Comment (KrishnaIyer) [reply DEVELOPMENT.md]: Added a reference to it instead.

**Pull request merged**